src/arrayMethodReduce.js

```diff
@@ -5,7 +5,21 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result;
+    let startIndex = 0;
```

Comment:
there is better use ternary operator
if an initial value exists return 0. If not - return 1